### PR TITLE
Resolve `char` as a primitive even if there is a module in scope

### LINKED ISSUE #58699: intra-doc links to modules instead of primitives inside standard library
For example in the doc for [`str::len`](https://doc.rust-lang.org/nightly/std/primitive.str.html#method.len):

> This length is in bytes, not [`char`](https://doc.rust-lang.org/nightly/std/char/index.html)s or graphemes.

The `char` in the doc links to [char/index.html](https://doc.rust-lang.org/nightly/std/char/index.html) instead of [primitive.char.html](https://doc.rust-lang.org/nightly/std/primitive.char.html).

Outside of the standard library, `char` links to the primitive, unless there is a `use std::char;` statement in which case it links to the module.

### PR DESCRIPTION
**Pull request by jyn514**
Closes https://github.com/rust-lang/rust/issues/58699.
r? @Manishearth

### CODE REVIEW

File: src/librustdoc/passes/collect_intra_doc_links.rs [hunk 1]
```diff
@@ -217,15 +216,12 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
...
                     Res::Def(DefKind::Mod, _) => {
-                        // This resolved to a module, but if we were passed `type@`,
-                        // we want primitive types to take precedence instead.
-                        if disambiguator == Some(Disambiguator::Namespace(Namespace::TypeNS)) {
-                            if let Some(prim) = is_primitive(path_str, ns) {
-                                if extra_fragment.is_some() {
-                                    return Err(ErrorKind::AnchorFailure(AnchorFailure::Primitive));
-                                }
-                                return Ok((prim, Some(path_str.to_owned())));
+                        // This resolved to a module, but we want primitive types to take precedence instead.
```
Comment (Manishearth) [inline src/librustdoc/passes/collect_intra_doc_links.rs:219]: We should avoid this branch if it's explicitly not specified as a type, yes?

Comment (jyn514) [reply src/librustdoc/passes/collect_intra_doc_links.rs:219]: Oh you're right - this should happen if there's no disambiguator or if the disambiguator is `type@`, but if there's a different ambiguator we should skip it.

Comment (jyn514) [reply src/librustdoc/passes/collect_intra_doc_links.rs:220]: Should be fixed now.

**Review by Manishearth — APPROVED** [review]

**Pull request merged**